src/node/jsonrpc.rs

```diff
@@ -0,0 +1,149 @@
+use crate::node::config::{
+    IPCAgentJsonRPCNodeConfig, DEFAULT_JSON_RPC_ENDPOINT, DEFAULT_JSON_RPC_VERSION,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::convert::Infallible;
+use warp::Filter;
+
+/// The IPC JSON RPC node that contains all the methods and handlers. The underlying implementation
+/// is using `warp`.
+///
+/// Note that currently only http json rpc is supported.
+///
+/// # Examples
+/// ```no_run
+/// use agent::node::config::IPCAgentJsonRPCNodeConfig;
+/// use agent::node::jsonrpc::IPCAgentJsonRPCNode;
+///
+/// #[tokio::main]
+/// async fn main() {
+///     let n = IPCAgentJsonRPCNode::new(IPCAgentJsonRPCNodeConfig::default());
+///     n.run().await;
+/// }
+/// ```
+pub struct IPCAgentJsonRPCNode {
+    config: IPCAgentJsonRPCNodeConfig,
+}
+
+impl IPCAgentJsonRPCNode {
+    pub fn new(config: IPCAgentJsonRPCNodeConfig) -> Self {
+        Self { config }
+    }
+
+    /// Runs the node in the current thread
+    pub async fn run(&self) {
+        log::info!("rpc node started at {:?}", self.config.addr());
+        warp::serve(json_rpc_filter()).run(self.config.addr()).await;
+    }
+}
+
+// Internal implementations
+
+/// Create the json_rpc filter
```

Comment:
Can you document what this filter does: 
- Listen to POST requests on the `DEFAULT_JSON_RPC_ENDPOINT` 
- Extract the body of the request. 
- Pass it to to the process function.

Reply:
updated and resolving this.